src/index.js

```diff
@@ -425,27 +432,67 @@ const registerLoginRoute = (keycloak) => {
     });
 };
 
+const registerDropSessionMethod = (keycloak) => {
+    keycloak.server.method('dropSession', (sessionId) => {
+        const cache = keycloak.server._core.caches.get(keycloak.config.cacheName);
+        if (cache) {
+            return cache.client.drop({id: sessionId, segment: '!yar'})
+        }
+        throw new Error(`Cannot find the "${keycloak.config.cacheName}" cache for drop procedure.`);
+    })
+};
+
+const registerBackChannelLogoutRoute = (keycloak) => {
+    keycloak.server.route({
+        path: keycloak.config.backChannelLogoutUrl,
+        method: 'POST',
+        handler: async (request, reply) => {
+            keycloak.server.log(['debug', 'keycloak'], 'Back-channel logout');
+            const logoutToken = new Token(request.payload);
+            const sessionIds = logoutToken.content.adapterSessionIds || [];
+            try {
+                await Promise.all(sessionIds.map(keycloak.server.methods.dropSession));
+            } catch (ex) {
+                keycloak.server.log(['warn', 'keycloak'], `An error occurred during dropping sessions. Error: ${ex}`);
+            }
+            return keycloak.answer(reply).representation('Successfully dropped all user\'s sessions.');
```

Comment:
This lacks error handling and cache control directives.
According to the specs:
- If the logout succeeded, the RP MUST respond with HTTP 200 OK. 
- If the logout request was invalid, the RP MUST respond with HTTP 400 Bad Request. 
- If the logout failed, the RP MUST respond with 501 Not Implemented. 
- If the local logout succeeded but some downstream logouts have failed, the RP MUST respond with HTTP 504 Gateway Timeout.
- The RP's response SHOULD include Cache-Control directives keeping the response from being cached to prevent cached responses from interfering with future logout requests. The following response headers should be included.
```
  Cache-Control: no-cache, no-store
  Pragma: no-cache
```